main.go

```diff
@@ -1,54 +1,84 @@
 package main
 
 import (
+	"container/list"
 	"io/ioutil"
-	"reflect"
-	"syscall/js"
+	"strconv"
+	"strings"
 
 	"github.com/rhysd/actionlint"
 )
 
-func toMap(input interface{}) map[string]interface{} {
-  out := make(map[string]interface{})
-  value := reflect.ValueOf(input)
-  if value.Kind() == reflect.Ptr {
-    value = value.Elem()
-  }
+var Memory = list.New()
+var OutputString = []byte{}
 
-  for i := 0; i < value.NumField(); i++ {
-    out[value.Type().Field(i).Name] = value.Field(i).Interface()
-  }
+type block struct {
+	ptr	*[]byte
+	value []byte
+}
+
+//export WasmAlloc
+func WasmAlloc(size int) *[]byte {
+	slice := make([]byte, size)
+	block := block{
+		ptr: &slice,
+		value: slice,
+	}
+	Memory.PushBack(block)
 
-  return out
+	return block.ptr
 }
 
-func runActionlint(source string, filePath string) (interface{}, error) {
+//export WasmFree
+func WasmFree(ptr *[]byte) {
+	for e := Memory.Front(); e != nil; e = e.Next() {
+		block := e.Value.(block)
+		if block.ptr == ptr {
+			Memory.Remove(e)
+			return
+		}
+	}
+}
+
+func serialize(errors []*actionlint.Error, target *[]byte) {
+	*target = []byte("[")
+
+	for i, err := range errors {
+		*target = append(*target, `{
+	"file":"`+err.Filepath+`",
+	"line":`+strconv.FormatInt(int64(err.Line), 10)+`,
+	"column":`+strconv.FormatInt(int64(err.Column), 10)+`,
+	"message":"`+strings.ReplaceAll(err.Message, `"`, `\"`)+`",
+	"kind":"`+strings.ReplaceAll(err.Kind, `"`, `\"`)+`"
+}`...)
+
+		if i < len(errors)-1 {
+			*target = append(*target, ',')
+		}
+	}
+
+	*target = append(*target, ']', 0)
```

Comment:
Any reason to not use https://pkg.go.dev/encoding/json here? It would be more readable

Reply:
It's not fully supported: https://github.com/tinygo-org/tinygo/issues/2660
Plus the above generates smaller output ;)